appservice/src/index.ts

```diff
@@ -3,4 +3,10 @@
  *  Licensed under the MIT License. See License.txt in the project root for license information.
  *--------------------------------------------------------------------------------------------*/
 
+export * from './AzureAccountWrapper';
```

Comment:
I think we should keep the 'api' surface as minimal as possible. Any change to any of the types you added becomes a breaking change and could cause issues. Ideally we just have one or two functions exported:
```
createAppService(credentials: ServiceClientCredentials, subscriptionId: string, appKind: AppKind)
```

or

```
createWebApp(credentials: ServiceClientCredentials, subscriptionId: string)
createFunctionApp(credentials: ServiceClientCredentials, subscriptionId: string)
```

You might need a few more parameters than that (a Memento for the persistence?) - but you get my point

Reply:
I like that, will do.